Added i18n translations to tile components

dwnoble opened the pull request:
* Added `intl. formatMessage` calls to hardcoded strings in tile components
* Re-generated client-side translation strings files
* Added locale parameter to `/api/place/displayname` calls. (Still todo: Add locale param support to /api/place/name)
* Updated place page locale string extractor (chart_config_extractor.py) to pull chart titles from new place page config
* Fixed issue where extract_messages.sh was throwing an error when trying to read in .d.ts typescript files

gmechali reviewed (approved):
Shoudl we add a webdriver test here in a different language to ensure the language replacement happens? I can imagine a couple ways how we could regress that.

beets commented on static/js/i18n/strings/en/place.json:
```diff
@@ -3,6 +3,22 @@
+  },
+  "chart-title-with-two-variables-and-location": {
+    "defaultMessage": "{variable1} Vs. {variable2} in {placeType} of {place}",
+    "description": "Chart title for a chart comparing two different variables, for places of a specific type within a place. For example, this could be Obesity Rate Vs. Income in States of USA, or Housing vs Poverty in Countries of Europe."
```
would be good to clarify the example title with quotes.

dwnoble replied:
Done

beets commented on static/js/i18n/strings/en/place.json:
```diff
@@ -67,18 +99,42 @@
     "description": "Text for header or subheader of Overview charts on place pages."
   },
+  "key_demographics": {
+    "defaultMessage": "Key Demographics",
```
Should this be sentence case (along with some of these other titles below)

dwnoble replied:
Updated

beets reviewed (approved):
nice! thanks for adding these strings

dwnoble commented on static/js/i18n/strings/en/place.json:
```diff
@@ -263,8 +343,40 @@
+  },
+  "viaGoogle": {
+    "defaultMessage": "via Google",
+    "description": "Indicator displayed when chart data comes from Google"
```
Done

dwnoble commented:
> Shoudl we add a webdriver test here in a different language to ensure the language replacement happens? I can imagine a couple ways how we could regress that unintentionally

Added a basic test, and added a TODO to update some of the assertions once we have actual translations in place
